lambda/getLocation/index.js

```diff
@@ -24,179 +16,183 @@ const dynamo = new AWS.DynamoDB.DocumentClient();
-    let statusCode = '200';
     const headers = {
         'Content-Type': 'application/json',
     };
-    
-    let parkingLotNameFilter = null
-    let recreationAreaFilter = null
-    let countyFilter = null
-    let parkUrlFilter = null
-    
-    if (event.queryStringParameters !== null && event.queryStringParameters !== undefined) {
-        parkingLotNameFilter = event.queryStringParameters.ParkingLotName
-        recreationAreaFilter = event.queryStringParameters.RecreationArea
-        countyFilter = event.queryStringParameters.CountyURL
-        parkUrlFilter = event.queryStringParameters.ParkURL
-    }
 
-
-    try {
-        body = await dynamo.scan({ 
-            TableName: "Locations"
-        }).promise();
-        
-        body = body.Items;
-        
-        let parkingData = await getParkingData()
-        
-        console.log(parkingData)
-        
-        if (parkingLotNameFilter !== null && parkingLotNameFilter !== undefined){
-            body = body.filter(location => location.ParkingLotName == parkingLotNameFilter)
-            body = body.map(item => {
-                item = {...item, ParkingData:
-                                        parkingData
-                                        .sort((a, b) => b.LastUpdate - a.LastUpdate)
-                                        .filter(parkingLot => parkingLot.ParkingLotId == item.Id)}
-                return item
-            })
-        } 
-        else if (parkUrlFilter !== null && parkUrlFilter !== undefined){
-            body = body.filter(location => location.ParkURL == parkUrlFilter);
-            body = body.map(item => {
-                item = {...item, ParkingData: 
-                                    parkingData
-                                    .sort((a, b) => b.LastUpdate - a.LastUpdate)
-                                    .filter(parkingLot => parkingLot.ParkingLotId == item.Id)}
-                return item
-            })
+    if (event.queryStringParameters && validate(event.queryStringParameters)) {
+        const query = event.queryStringParameters
         
-            body = {RecreationArea: body[0].RecreationArea,
-                    About: body[0].About,
-                    Images: body[0].Images,
-                    List : body
-                    }
-                    
-
-        } 
-        else if (recreationAreaFilter !== null && recreationAreaFilter !== undefined){
-            body = body.filter(location => location.RecreationArea == recreationAreaFilter)
-            body = body.map(item => {
-                item = {...item, ParkingData: 
+        try {
+            result = await dynamo.scan({ 
+                TableName: "Locations"
+            }).promise();
+            
+            result = result.Items;
+            
+            let parkingData = await getParkingData()
+            
+            console.log(parkingData)
+            
+            if (query.ParkingLotName){ // Get location data by parking lot name
```

Comment:
TODO: Pull this out into separate functions that can be called here. User DB filters instead of iterating over objects like this
<img width="863" alt="168934235-20442acb-7b09-40bd-8efe-0237c9c09f1e" src="https://user-images.githubusercontent.com/16601367/169160781-cf725fd7-95c9-4bd4-a32c-78b997c4d9a1.png">